src/microbe_stage/organelle_unlocks/ConditionSet.cs

```diff
@@ -0,0 +1,72 @@
+﻿namespace UnlockConstraints
+{
+    using System.Linq;
+    using Newtonsoft.Json;
+
+    [JsonConverter(typeof(ConditionSetConverter))]
+    public class ConditionSet
+    {
+        public ConditionSet(IUnlockCondition[] conditions)
+        {
+            Requirements = conditions;
+        }
+
+        [JsonIgnore]
+        public IUnlockCondition[] Requirements { get; set; }
+
+        public bool Satisfied(WorldAndPlayerEventArgs worldAndPlayerArgs)
+        {
+            var statisticArgs = new StatisticTrackerEventArgs(worldAndPlayerArgs.World.StatisticsTracker);
+
+            return Requirements.All(c => IsSatisfied(c, worldAndPlayerArgs, statisticArgs));
+        }
+
+        public void GenerateTooltip(LocalizedStringBuilder builder, WorldAndPlayerEventArgs worldAndPlayerArgs)
+        {
+            var statisticArgs = new StatisticTrackerEventArgs(worldAndPlayerArgs.World.StatisticsTracker);
+
+            var first = true;
+            foreach (var entry in Requirements)
+            {
+                if (!first)
+                {
+                    builder.Append(new LocalizedString("AND_UNLOCK_CONDITION"));
+                    builder.Append(" ");
+                }
+
+                var color = IsSatisfied(entry, worldAndPlayerArgs, statisticArgs) ? "green" : "red";
```

Comment:
These should use colour codes and be defined in Constants.cs. Also the pure red and green are likely way too strong colours and should be diluted a bit to match the other reds and greens used in the game.